From 220c29d0e10e4a869bffe1f0d4a54e233020d9dd Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Thu, 31 May 2012 16:00:34 +0200
Subject: [PATCH] man: add documentation for the binfmt, modules-load, sysctl
 services (cherry picked from commit
 9393a8774c1acd60deea40007061b9ffc783bf7e)

Conflicts:
	Makefile.am
---
 Makefile.am                           |   16 ++++++-
 man/binfmt.d.xml                      |   53 +++++++++++++---------
 man/modules-load.d.xml                |   52 ++++++++++++----------
 man/sysctl.d.xml                      |   30 +++++++------
 man/systemd-binfmt.service.xml        |   76 ++++++++++++++++++++++++++++++++
 man/systemd-modules-load.service.xml  |   76 ++++++++++++++++++++++++++++++++
 man/systemd-sysctl.service.xml        |   78 +++++++++++++++++++++++++++++++++
 units/systemd-binfmt.service.in       |    3 +-
 units/systemd-modules-load.service.in |    2 +-
 units/systemd-sysctl.service.in       |    2 +-
 10 files changed, 328 insertions(+), 60 deletions(-)
 create mode 100644 man/systemd-binfmt.service.xml
 create mode 100644 man/systemd-modules-load.service.xml
 create mode 100644 man/systemd-sysctl.service.xml

diff --git a/Makefile.am b/Makefile.am
index 86adad0..600becf 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -467,7 +467,9 @@ MANPAGES = \
 	man/os-release.5 \
 	man/machine-info.5 \
 	man/modules-load.d.5 \
+	man/systemd-modules-load.service.8 \
 	man/sysctl.d.5 \
+	man/systemd-sysctl.service.8 \
 	man/systemd-ask-password.1 \
 	man/systemd-cat.1 \
 	man/systemd-machine-id-setup.1 \
@@ -482,7 +484,9 @@ MANPAGES_ALIAS = \
 	man/init.1 \
 	man/systemd-systemctl.1 \
 	man/systemd-loginctl.1 \
-	man/systemd-journalctl.1
+	man/systemd-journalctl.1 \
+	man/systemd-modules-load.8 \
+	man/systemd-sysctl.8
 
 man/reboot.8: man/halt.8
 man/poweroff.8: man/halt.8
@@ -490,6 +494,8 @@ man/init.1: man/systemd.1
 man/systemd-systemctl.1: man/systemctl.1
 man/systemd-loginctl.1: man/loginctl.1
 man/systemd-journalctl.1: man/journalctl.1
+man/systemd-modules-load.8: man/systemd-modules-load.service.8
+man/systemd-sysctl.8: man/systemd-sysctl.service.8
 
 XML_FILES = \
 	${patsubst %.1,%.xml,${patsubst %.3,%.xml,${patsubst %.5,%.xml,${patsubst %.7,%.xml,${patsubst %.8,%.xml,$(MANPAGES)}}}}}
@@ -1682,7 +1688,13 @@ INSTALL_DATA_HOOKS += \
 	binfmt-install-data-hook
 
 MANPAGES += \
-	man/binfmt.d.5
+	man/binfmt.d.5 \
+	man/systemd-binfmt.service.8
+
+MANPAGES_ALIAS +=  \
+	man/systemd-binfmt.8
+
+man/systemd-binfmt.8: man/systemd-binfmt.service.8
 
 EXTRA_DIST += \
 	units/systemd-binfmt.service.in
diff --git a/man/binfmt.d.xml b/man/binfmt.d.xml
index ca40ad9..fedde88 100644
--- a/man/binfmt.d.xml
+++ b/man/binfmt.d.xml
@@ -42,7 +42,8 @@
 
         <refnamediv>
                 <refname>binfmt.d</refname>
-                <refpurpose>Configure additional binary formats at boot</refpurpose>
+                <refpurpose>Configure additional binary formats for
+                executables at boot</refpurpose>
         </refnamediv>
 
         <refsynopsisdiv>
@@ -54,10 +55,11 @@
         <refsect1>
                 <title>Description</title>
 
-		<para><command>systemd</command> uses
-		files from the above directories to configure
-		additional binary formats to register during boot in
-		the kernel.</para>
+		<para>At boot,
+		<citerefentry><refentrytitle>systemd-binfmt.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>
+		reads configuration files from the above directories
+		to register in the kernel additional binary
+		formats for executables.</para>
         </refsect1>
 
         <refsect1>
@@ -73,21 +75,30 @@
                 ignored. Note that this means you may not use ; and #
                 as delimiter in binary format rules.</para>
 
-                <para>Each configuration file is named in the style of
-                <filename>&lt;program&gt;.conf</filename>.
-                Files in <filename>/etc/</filename> override
-                files with the same name in <filename>/usr/lib/</filename>.
-                Files in <filename>/run</filename> override files with
-                the same name in <filename>/etc/</filename> and
-                <filename>/usr/lib/</filename>. Packages should install their
-                configuration files in <filename>/usr/lib/</filename>, files
-                in <filename>/etc/</filename> are reserved for the local
-                administration, which possibly decides to override the
-                configurations installed from packages. All files are sorted
-                by filename in alphabetical order, regardless in which of the
-                directories they reside, to ensure that a specific
-                configuration file takes precedence over another file with
-                an alphabetically later name.</para>
+                <para>Each configuration file shall be named in the
+                style of <filename>&lt;program&gt;.conf</filename>.
+                Files in <filename>/etc/</filename> override files
+                with the same name in <filename>/usr/lib/</filename>
+                and <filename>/run/</filename>. Files in
+                <filename>/run/</filename> override files with the
+                same name in <filename>/usr/lib/</filename>. Packages
+                should install their configuration files in
+                <filename>/usr/lib/</filename>, files in
+                <filename>/etc/</filename> are reserved for the local
+                administrator, who may use this logic to override the
+                configuration files installed from vendor
+                packages. All files are sorted by their filename in
+                alphabetical order, regardless in which of the
+                directories they reside, to guarantee that a specific
+                configuration file takes precedence over another file
+                with an alphabetically later name.</para>
+
+                <para>If the administrator wants to disable a
+                configuration file supplied by the vendor the
+                recommended way is to place a symlink to
+                <filename>/dev/null</filename> in
+                <filename>/etc/binfmt.d/</filename> bearing the
+                same file name.</para>
         </refsect1>
 
         <refsect1>
@@ -104,6 +115,8 @@
                 <title>See Also</title>
                 <para>
                         <citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry>,
+                        <citerefentry><refentrytitle>systemd-binfmt.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
+                        <citerefentry><refentrytitle>systemd-delta</refentrytitle><manvolnum>1</manvolnum></citerefentry>,
                         <citerefentry><refentrytitle>wine</refentrytitle><manvolnum>8</manvolnum></citerefentry>
                 </para>
         </refsect1>
diff --git a/man/modules-load.d.xml b/man/modules-load.d.xml
index 9b1afc9..9ae5020 100644
--- a/man/modules-load.d.xml
+++ b/man/modules-load.d.xml
@@ -54,16 +54,17 @@
         <refsect1>
                 <title>Description</title>
 
-		<para><command>systemd</command> uses
-		files from the above directories to configure
+		<para><citerefentry><refentrytitle>systemd-modules-load.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>
+		reads files from the above directories which contain
 		kernel modules to load during boot in a static list.
 		Each configuration file is named in the style of
 		<filename>/etc/modules-load.d/&lt;program&gt;.conf</filename>. Note
-		that it is usually a better idea to use the automatic
-		module loading by PCI ID, by DMI ID or similar
-		triggers configured in the kernel modules themselves
-		instead of relying on static configuration like
-		this.</para>
+		that it is usually a better idea to rely on the
+		automatic module loading by PCI IDs, USB IDs, DMI IDs
+		or similar triggers encoded in the kernel modules
+		themselves instead of static configuration like
+		this. In fact, most modern kernel modules are prepared
+		for automatic loading already.</para>
         </refsect1>
 
         <refsect1>
@@ -74,21 +75,26 @@
 		newlines. Empty lines and lines whose first
 		non-whitespace character is # or ; are ignored.</para>
 
-                <para>Each configuration file is named in the style of
-                <filename>&lt;program&gt;.conf</filename>.
-                Files in <filename>/etc/</filename> override
-                files with the same name in <filename>/usr/lib/</filename>.
-                Files in <filename>/run</filename> override files with
-                the same name in <filename>/etc/</filename> and
-                <filename>/usr/lib/</filename>. Packages should install their
-                configuration files in <filename>/usr/lib/</filename>, files
-                in <filename>/etc/</filename> are reserved for the local
-                administration, which possibly decides to override the
-                configurations installed from packages. All files are sorted
-                by filename in alphabetical order, regardless in which of the
-                directories they reside, to ensure that a specific
-                configuration file takes precedence over another file with
-                an alphabetically later name.</para>
+                <para>Each configuration file shall be named in the
+                style of <filename>&lt;program&gt;.conf</filename>.
+                Files in <filename>/etc/</filename> override files
+                with the same name in <filename>/usr/lib/</filename>
+                and <filename>/run/</filename>. Files in
+                <filename>/run/</filename> override files with the
+                same name in <filename>/usr/lib/</filename>. Packages
+                should install their configuration files in
+                <filename>/usr/lib/</filename>, files in
+                <filename>/etc/</filename> are reserved for the local
+                administrator, who may use this logic to override the
+                configuration files installed from vendor
+                packages.</para>
+
+                <para>If the administrator wants to disable a
+                configuration file supplied by the vendor the
+                recommended way is to place a symlink to
+                <filename>/dev/null</filename> in
+                <filename>/etc/modules-load.d/</filename> bearing the
+                same file name.</para>
         </refsect1>
 
         <refsect1>
@@ -105,6 +111,8 @@ virtio-net</programlisting>
                 <title>See Also</title>
                 <para>
                         <citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry>,
+                        <citerefentry><refentrytitle>systemd-modules-load.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
+                        <citerefentry><refentrytitle>systemd-delta</refentrytitle><manvolnum>1</manvolnum></citerefentry>,
                         <citerefentry><refentrytitle>modprobe</refentrytitle><manvolnum>8</manvolnum></citerefentry>
                 </para>
         </refsect1>
diff --git a/man/sysctl.d.xml b/man/sysctl.d.xml
index 20f2e24..f6dd52f 100644
--- a/man/sysctl.d.xml
+++ b/man/sysctl.d.xml
@@ -54,10 +54,12 @@
         <refsect1>
                 <title>Description</title>
 
-		<para><command>systemd</command> uses configuration
-		files from the above directories to configure
+		<para>At boot,
+		<citerefentry><refentrytitle>systemd-binfmt.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>
+		reads configuration files from the above directories
+		to configure
 		<citerefentry><refentrytitle>sysctl</refentrytitle><manvolnum>8</manvolnum></citerefentry>
-		kernel parameters during boot.</para>
+		kernel parameters.</para>
         </refsect1>
 
         <refsect1>
@@ -76,17 +78,17 @@
 
                 <para>Each configuration file shall be named in the
                 style of <filename>&lt;program&gt;.conf</filename>.
-                Files in <filename>/run/</filename> override files
-                with the same name in <filename>/usr/lib/</filename>.
-                Files in <filename>/etc</filename> override files with
-                the same name in <filename>/run/</filename> and
-                <filename>/usr/lib/</filename>. Packages should
-                install their configuration files in
+                Files in <filename>/etc/</filename> override files
+                with the same name in <filename>/usr/lib/</filename>
+                and <filename>/run/</filename>.  Files in
+                <filename>/run</filename> override files with the same
+                name in <filename>/usr/lib/</filename>. Packages
+                should install their configuration files in
                 <filename>/usr/lib/</filename>. Files in
                 <filename>/etc/</filename> are reserved for the local
                 administrator, who may use this logic to override the
-                configuration installed by vendor packages. All
-                configuration files are sorted by their name in
+                configuration files installed by vendor packages. All
+                configuration files are sorted by their filename in
                 alphabetical order, regardless in which of the
                 directories they reside, to guarantee that a specific
                 configuration file takes precedence over another file
@@ -97,8 +99,8 @@
                 configuration file supplied by the vendor the
                 recommended way is to place a symlink to
                 <filename>/dev/null</filename> in
-                <filename>/etc/sysctl.d</filename> carrying with the
-                same name.</para>
+                <filename>/etc/sysctl.d/</filename> bearing the
+                same file name.</para>
         </refsect1>
 
         <refsect1>
@@ -115,6 +117,8 @@ kernel.domainname=example.com</programlisting>
                 <title>See Also</title>
                 <para>
                         <citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry>,
+                        <citerefentry><refentrytitle>systemd-sysctl.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
+                        <citerefentry><refentrytitle>systemd-delta</refentrytitle><manvolnum>1</manvolnum></citerefentry>,
                         <citerefentry><refentrytitle>sysctl</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
                         <citerefentry><refentrytitle>sysctl.conf</refentrytitle><manvolnum>5</manvolnum></citerefentry>
                 </para>
diff --git a/man/systemd-binfmt.service.xml b/man/systemd-binfmt.service.xml
new file mode 100644
index 0000000..73e9836
--- /dev/null
+++ b/man/systemd-binfmt.service.xml
@@ -0,0 +1,76 @@
+<?xml version="1.0"?>
+<!--*-nxml-*-->
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN" "http://www.oasis-open.org/docbook/xml/4.2/docbookx.dtd">
+<!--
+  This file is part of systemd.
+
+  Copyright 2012 Lennart Poettering
+
+  systemd is free software; you can redistribute it and/or modify it
+  under the terms of the GNU Lesser General Public License as published by
+  the Free Software Foundation; either version 2.1 of the License, or
+  (at your option) any later version.
+
+  systemd is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with systemd; If not, see <http://www.gnu.org/licenses/>.
+-->
+<refentry id="systemd-binfmt.service">
+
+        <refentryinfo>
+                <title>systemd-binfmt.service</title>
+                <productname>systemd</productname>
+
+                <authorgroup>
+                        <author>
+                                <contrib>Developer</contrib>
+                                <firstname>Lennart</firstname>
+                                <surname>Poettering</surname>
+                                <email>lennart@poettering.net</email>
+                        </author>
+                </authorgroup>
+        </refentryinfo>
+
+        <refmeta>
+                <refentrytitle>systemd-binfmt.service</refentrytitle>
+                <manvolnum>8</manvolnum>
+        </refmeta>
+
+        <refnamediv>
+                <refname>systemd-binfmt.service</refname>
+                <refname>systemd-binfmt</refname>
+                <refpurpose>Configure additional binary formats for executables at boot</refpurpose>
+        </refnamediv>
+
+        <refsynopsisdiv>
+                <para><filename>sysytemd-binfmt.service</filename></para>
+                <para><filename>/usr/lib/systemd/systemd-binfmt</filename></para>
+        </refsynopsisdiv>
+
+        <refsect1>
+                <title>Description</title>
+
+		<para><filename>systemd-binfmt.service</filename> is
+		an early-boot service that registers additional binary
+		formats for executables in the kernel.</para>
+
+                <para>See
+                <citerefentry><refentrytitle>binfmt.d</refentrytitle><manvolnum>5</manvolnum></citerefentry>
+                for information about the configuration of this
+                service.</para>
+        </refsect1>
+
+        <refsect1>
+                <title>See Also</title>
+                <para>
+                        <citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry>,
+                        <citerefentry><refentrytitle>binfmt.d</refentrytitle><manvolnum>5</manvolnum></citerefentry>,
+                        <citerefentry><refentrytitle>wine</refentrytitle><manvolnum>8</manvolnum></citerefentry>
+                </para>
+        </refsect1>
+
+</refentry>
diff --git a/man/systemd-modules-load.service.xml b/man/systemd-modules-load.service.xml
new file mode 100644
index 0000000..405ae4f
--- /dev/null
+++ b/man/systemd-modules-load.service.xml
@@ -0,0 +1,76 @@
+<?xml version="1.0"?>
+<!--*-nxml-*-->
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN" "http://www.oasis-open.org/docbook/xml/4.2/docbookx.dtd">
+<!--
+  This file is part of systemd.
+
+  Copyright 2012 Lennart Poettering
+
+  systemd is free software; you can redistribute it and/or modify it
+  under the terms of the GNU Lesser General Public License as published by
+  the Free Software Foundation; either version 2.1 of the License, or
+  (at your option) any later version.
+
+  systemd is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with systemd; If not, see <http://www.gnu.org/licenses/>.
+-->
+<refentry id="systemd-modules-load.service">
+
+        <refentryinfo>
+                <title>systemd-modules-load.service</title>
+                <productname>systemd</productname>
+
+                <authorgroup>
+                        <author>
+                                <contrib>Developer</contrib>
+                                <firstname>Lennart</firstname>
+                                <surname>Poettering</surname>
+                                <email>lennart@poettering.net</email>
+                        </author>
+                </authorgroup>
+        </refentryinfo>
+
+        <refmeta>
+                <refentrytitle>systemd-modules-load.service</refentrytitle>
+                <manvolnum>8</manvolnum>
+        </refmeta>
+
+        <refnamediv>
+                <refname>systemd-modules-load.service</refname>
+                <refname>systemd-modules-load</refname>
+                <refpurpose>Configure kernel modules to load at boot</refpurpose>
+        </refnamediv>
+
+        <refsynopsisdiv>
+                <para><filename>sysytemd-modules-load.service</filename></para>
+                <para><filename>/usr/lib/systemd/systemd-modules-load</filename></para>
+        </refsynopsisdiv>
+
+        <refsect1>
+                <title>Description</title>
+
+		<para><filename>systemd-modules-load.service</filename>
+		is an early-boot service that loads kernel modules
+		from static configuration.</para>
+
+                <para>See
+                <citerefentry><refentrytitle>modules-load.d</refentrytitle><manvolnum>5</manvolnum></citerefentry>
+                for information about the configuration of this
+                service.</para>
+        </refsect1>
+
+        <refsect1>
+                <title>See Also</title>
+                <para>
+                        <citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry>,
+                        <citerefentry><refentrytitle>modules-load.d</refentrytitle><manvolnum>5</manvolnum></citerefentry>,
+                        <citerefentry><refentrytitle>wine</refentrytitle><manvolnum>8</manvolnum></citerefentry>
+                </para>
+        </refsect1>
+
+</refentry>
diff --git a/man/systemd-sysctl.service.xml b/man/systemd-sysctl.service.xml
new file mode 100644
index 0000000..116fcd0
--- /dev/null
+++ b/man/systemd-sysctl.service.xml
@@ -0,0 +1,78 @@
+<?xml version="1.0"?>
+<!--*-nxml-*-->
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN" "http://www.oasis-open.org/docbook/xml/4.2/docbookx.dtd">
+<!--
+  This file is part of systemd.
+
+  Copyright 2012 Lennart Poettering
+
+  systemd is free software; you can redistribute it and/or modify it
+  under the terms of the GNU Lesser General Public License as published by
+  the Free Software Foundation; either version 2.1 of the License, or
+  (at your option) any later version.
+
+  systemd is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with systemd; If not, see <http://www.gnu.org/licenses/>.
+-->
+<refentry id="systemd-sysctl.service">
+
+        <refentryinfo>
+                <title>systemd-sysctl.service</title>
+                <productname>systemd</productname>
+
+                <authorgroup>
+                        <author>
+                                <contrib>Developer</contrib>
+                                <firstname>Lennart</firstname>
+                                <surname>Poettering</surname>
+                                <email>lennart@poettering.net</email>
+                        </author>
+                </authorgroup>
+        </refentryinfo>
+
+        <refmeta>
+                <refentrytitle>systemd-sysctl.service</refentrytitle>
+                <manvolnum>8</manvolnum>
+        </refmeta>
+
+        <refnamediv>
+                <refname>systemd-sysctl.service</refname>
+                <refname>systemd-sysctl</refname>
+                <refpurpose>Configure kernel parameters at boot</refpurpose>
+        </refnamediv>
+
+        <refsynopsisdiv>
+                <para><filename>sysytemd-sysctl.service</filename></para>
+                <para><filename>/usr/lib/systemd/systemd-sysctl</filename></para>
+        </refsynopsisdiv>
+
+        <refsect1>
+                <title>Description</title>
+
+		<para><filename>systemd-sysctl.service</filename> is
+		an early-boot service that configures
+		<citerefentry><refentrytitle>sysctl</refentrytitle><manvolnum>8</manvolnum></citerefentry>
+		kernel parameters.</para>
+
+                <para>See
+                <citerefentry><refentrytitle>sysctl.d</refentrytitle><manvolnum>5</manvolnum></citerefentry>
+                for information about the configuration of this
+                service.</para>
+        </refsect1>
+
+        <refsect1>
+                <title>See Also</title>
+                <para>
+                        <citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry>,
+                        <citerefentry><refentrytitle>sysctl.d</refentrytitle><manvolnum>5</manvolnum></citerefentry>,
+		        <citerefentry><refentrytitle>sysctl</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
+                        <citerefentry><refentrytitle>wine</refentrytitle><manvolnum>8</manvolnum></citerefentry>
+                </para>
+        </refsect1>
+
+</refentry>
diff --git a/units/systemd-binfmt.service.in b/units/systemd-binfmt.service.in
index 1295ca0..27acabb 100644
--- a/units/systemd-binfmt.service.in
+++ b/units/systemd-binfmt.service.in
@@ -7,13 +7,14 @@
 
 [Unit]
 Description=Set Up Additional Binary Formats
-Documentation=man:binfmt.d(5)
+Documentation=man:systemd-binfmt.service(8) man:binfmt.d(5)
 Documentation=https://www.kernel.org/doc/Documentation/binfmt_misc.txt
 DefaultDependencies=no
 Conflicts=shutdown.target
 After=systemd-readahead-collect.service systemd-readahead-replay.service proc-sys-fs-binfmt_misc.automount
 Before=sysinit.target shutdown.target
 ConditionPathIsReadWrite=/proc/sys/
+ConditionDirectoryNotEmpty=|/lib/binfmt.d
 ConditionDirectoryNotEmpty=|/usr/lib/binfmt.d
 ConditionDirectoryNotEmpty=|/usr/local/lib/binfmt.d
 ConditionDirectoryNotEmpty=|/etc/binfmt.d
diff --git a/units/systemd-modules-load.service.in b/units/systemd-modules-load.service.in
index 6c2a164..90d1fe0 100644
--- a/units/systemd-modules-load.service.in
+++ b/units/systemd-modules-load.service.in
@@ -7,7 +7,7 @@
 
 [Unit]
 Description=Load Kernel Modules
-Documentation=man:modules-load.d(5)
+Documentation=man:systemd-modules-load.service(8) man:modules-load.d(5)
 DefaultDependencies=no
 Conflicts=shutdown.target
 After=systemd-readahead-collect.service systemd-readahead-replay.service
diff --git a/units/systemd-sysctl.service.in b/units/systemd-sysctl.service.in
index 4794df9..1ec1e79 100644
--- a/units/systemd-sysctl.service.in
+++ b/units/systemd-sysctl.service.in
@@ -7,7 +7,7 @@
 
 [Unit]
 Description=Apply Kernel Variables
-Documentation=man:sysctl.d(5)
+Documentation=man:systemd-sysctl.service(8) man:sysctl.d(5)
 DefaultDependencies=no
 Conflicts=shutdown.target
 After=systemd-readahead-collect.service systemd-readahead-replay.service
